src/Data/Record/Extra.purs

```diff
@@ -175,6 +175,38 @@ class Applicative m <= SequenceRecord rl row row' m
   where
     sequenceRecordImpl :: RLProxy rl -> Record row -> m (Record row')
 
+class OrdRecord rl row
+  | rl -> row
+  where
+    compareRecordImpl :: RLProxy rl -> Record row -> Record row -> Ordering
+
+instance ordRecordCons ::
+  ( IsSymbol name
+  , Ord ty
+  , RowCons name ty trash row
+  , OrdRecord tail row
+  ) => OrdRecord (Cons name ty tail) row where
+  compareRecordImpl _ a b =
+    case compare valA valB of
```

Comment:
another just minor thing, i think this generates code that compares the whole thing, whereas `let result = __ if ___ then ___ else ___` will generate code that does not evaluate the whole branch? might require some digging in with `spy`

Reply:
Here is the generated code:

```purescript
                        return function (b) {
                            var valA = Data_Record.get(dictIsSymbol)(dictRowCons)(Data_Symbol.SProxy.value)(a);
                            var valB = Data_Record.get(dictIsSymbol)(dictRowCons)(Data_Symbol.SProxy.value)(b);
                            var v1 = Data_Ord.compare(dictOrd)(valA)(valB);
                            if (v1 instanceof Data_Ordering.EQ) {
                                return compareRecordImpl(dictOrdRecord)(Type_Row.RLProxy.value)(a)(b);
                            };
                            return v1;
                        };
```

Should be fine, no?

Reply:
oh, cool, seems fine then